data/service/api/v1/summary.go

```diff
@@ -0,0 +1,113 @@
+package v1
+
+import (
+	"net/http"
+    "strconv"
+
+	//"github.com/tidepool-org/platform/data"
+	dataService "github.com/tidepool-org/platform/data/service"
+	//"github.com/tidepool-org/platform/page"
+	//"github.com/tidepool-org/platform/permission"
+	"github.com/tidepool-org/platform/request"
+)
+
+func SummaryRoutes() []dataService.Route {
+	return []dataService.Route{
+		dataService.MakeRoute("GET", "/v1/summary/:userId", Authenticate(GetSummary)),
+		dataService.MakeRoute("POST", "/v1/summary/:userId", Authenticate(UpdateSummary)),
+		dataService.MakeRoute("GET", "/v1/agedsummaries/:minutes", Authenticate(GetAgedSummaries)),
+	}
+}
+
+func GetSummary(dataServiceContext dataService.Context) {
+	res := dataServiceContext.Response()
+	req := dataServiceContext.Request()
+	dataClient := dataServiceContext.DataClient()
+
+	//details := request.DetailsFromContext(req.Context())
+	/*if details == nil {
+		request.MustNewResponder(res, req).Error(http.StatusUnauthorized, request.ErrorUnauthenticated())
+		return
+	}*/
+
+	responder := request.MustNewResponder(res, req)
+
+	id := req.PathParam("userId")
+	if id == "" {
+		responder.Error(http.StatusBadRequest, request.ErrorParameterMissing("id"))
+		return
+	}
+
+	summary, err := dataClient.GetSummary(req.Context(), id)
+	if err != nil {
+		responder.Error(http.StatusInternalServerError, err)
+		return
+	}
+
+	responder.Data(http.StatusOK, summary)
+}
+
+
+func UpdateSummary(dataServiceContext dataService.Context) {
+	res := dataServiceContext.Response()
+	req := dataServiceContext.Request()
+	dataClient := dataServiceContext.DataClient()
+
+	//details := request.DetailsFromContext(req.Context())
+	/*if details == nil {
```

Comment:
Since this API is (presumably) only called by other services, the permissions check here would be to check that the token is a server token (i.e. `details.isService()`)